### PR TITLE
Py3

### PR DESCRIPTION
**Pull request by taylorcenters**
Merged SecureAB fork for py3 support + additional updates for py3 and py2 support

### CODE REVIEW

File: tools/zbscapy [hunk 1]
```diff
@@ -7,7 +7,7 @@ log_killerbee = logging.getLogger('scapy.killerbee')
 try:
 	from scapy.all import *
 except ImportError:
-	print 'This tool requires Scapy to be installed (specifically scapy-com edition).'
+	print('This tool requires Scapy to be installed (specifically scapy-com edition).')
```
Comment (rmspeers) [inline tools/zbscapy]: I think now regular scapy has the dot15d4 and zigbee layers included?

File: tools/zbrealign [hunk 2]
```diff
@@ -15,7 +15,7 @@ from killerbee import *
 try:
 	from scapy.all import *
 except ImportError:
-	print 'This tool requires Scapy to be installed (specifically scapy-com edition).'
+	print('This tool requires Scapy to be installed (specifically scapy-com edition).')
```
Comment (rmspeers) [inline tools/zbrealign]: Ditto below warning can be changed -- see https://github.com/secdev/scapy/tree/master/scapy/layers that both layers are now in trunk.

File: tools/zbstumbler [hunk 3]
```diff
@@ -41,20 +41,20 @@ def display_details(routerdata):
         sys.stdout.write("\tExt PANID: Unknown")
 
     try:
-        print "\tStack Profile: %s"%stackprofile_map[stackprofile]
+        print("\tStack Profile: %s"%stackprofile_map[stackprofile])
```
Comment (rmspeers) [inline tools/zbstumbler]: Should standardize these to all use .format() as most are already.

File: tools/zbstumbler [hunk 4]
```diff
@@ -101,7 +99,7 @@ def response_handler(stumbled, packet, channel):
         return value
 
     if args.verbose:
-        print("Received frame is not a beacon (FCF={0}).".format(pktdecode[0].encode('hex')))
+        print(("Received frame is not a beacon (FCF={0}).".format(pktdecode[0].encode('hex'))))
```
Comment (rmspeers) [inline tools/zbstumbler]: Does this .hex work in py3, or do you need `binascii.hexlify` now?

File: tools/zbstumbler [hunk 5]
```diff
@@ -171,10 +169,10 @@ if __name__ == '__main__':
 
         if not args.channel:
             if args.verbose:
-                print("Setting channel to {0}.".format(channel))
+                print(("Setting channel to {0}.".format(channel)))
             try:
                 kb.set_channel(channel)
-            except Exception, e:
+            except Exception as e:
                 print("ERROR: Failed to set channel to {0}. ({1})".format(channel, e))
```
Comment (rmspeers) [inline tools/zbstumbler]: send errors to stderr

**Review by rmspeers — COMMENTED** [review]
First syntax skim, functionality tests needed.

**Comment by SecureAB** [comment]
I used `py3` as my first draft and as a working copy and `py3-part1` as my `release` branch. So `py3` also contains commits which are not needed for a python3 port. It also includes commits which I used to test, if something would be a good way to go for a python3 port. I'm not sure if the changes in the `py3` branch are backwards compatible.

File: tools/zbstumbler (hunk 4 above)
Comment (taylorcenters) [reply tools/zbstumbler]: I believe you're correct, the hex encoding was removed in py3. recommended to convert string to bytes and use .hex()
will update

File: README.md [hunk 6]
```diff
@@ -15,6 +29,7 @@ The main toolkit was/is authored by:
 + 2010-2011, Ricky Melgares <ricky@riverloopsecurity.com>
 
 We appreciate the many contributers to the framework, including the following who have contributed capabilities:
++ Taylor Centers <taylor@riverloopsecurity.com> (Python 3 port)
```
Comment (rmspeers) [inline README.md]: Want to maybe give a shout out to other people who submitted py3 PRs?

**Review by rmspeers — APPROVED** [review]
Per notes in PR/Slack, I hear this works and has been tested. Approving as such into beta.

**Pull request merged**